resources/js/components/HasMany/HasMany.tsx

```diff
@@ -17,42 +17,86 @@ export const HasMany: React.VFC<{
 	}, [items]);
 
 	const updateItem = (index: number, value: string) => {
-		const newItems = [...items];
+		const newItems = JSON.parse(JSON.stringify(items));
 		newItems[index].value = value;
```

Comment:
Wouldn't this work?
```
const newItems = [...items];
newItems[index] = {...newItems[index], value};
```

Reply:
It would, but currently I don't see or am aware of any imediate benefit that your way would provide.